flatson/flatson.py

```diff
@@ -137,3 +124,20 @@ def flatten_dict(self, obj):
         """Return an OrderedDict dict preserving order of keys in fieldnames
         """
         return OrderedDict(zip(self.fieldnames, self.flatten(obj)))
+
+    def infer_flattened_field_names(self, schema, field_sep='.'):
```

Comment:
Yeap, that one is asking to be refactored! :+1:
I'd start with adding isolated tests, though.

That refactoring is interesting, I think that is the way to go.
I have some small easy-to-fix concerns about it, I'll comment in that PR.